### PR TITLE
Add io.itch.ksylvestre.MightyGrepDemo

### PR DESCRIPTION
**Pull request by kyle-sylvestre**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*

1. --filesystem=host: allow users to search more files on their system
2. --talk-name=org.freedesktop.Flatpak: allow users to use specify "Open in editor" and "Open file location" commands

- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

**Comment by kyle-sylvestre** [comment]
remade to fix branch name

**Comment by bbhtt** [comment]
At least for open file location, you should be using the OpenURI portal OpenDir method that works without holes https://flatpak.github.io/xdg-desktop-portal/docs/doc-org.freedesktop.portal.OpenURI.html or the org.freedesktop.Filemanager1 interface. How is that handled right now?

How is opening in external editor handled? Do you have a list of editors hardcoded?

File: io.itch.ksylvestre.MightyGrepDemo.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  - --talk-name=org.freedesktop.Flatpak
+modules:
+  - name: mightygrep-demo
+    buildsystem: simple
+    sources:
+      - type: archive
+        filename: flatpak-demo
+        url: https://github.com/kyle-sylvestre/mightygrep-misc/releases/download/0.9.1/flatpak-demo.tar.gz
+        sha256: c303e5db11015e1c8fd870770e75cc14d5613787f8d078a799bf8d4528d12087
+        size: 969470
```
Comment (hfiguiere) [inline io.itch.ksylvestre.MightyGrepDemo.yml:20]:
```suggestion
```

**Comment by hfiguiere** [comment]
bot, build io.itch.ksylvestre.MightyGrepDemo

File: flathub.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: so why this:
image

**Comment by kyle-sylvestre** [comment]
> At least for open file location, you should be using the OpenURI portal OpenDir method that works without holes https://flatpak.github.io/xdg-desktop-portal/docs/doc-org.freedesktop.portal.OpenURI.html or the org.freedesktop.Filemanager1 interface. How is that handled right now?
> 
> How is opening in external editor handled? Do you have a list of editors hardcoded?

yes I use a list of hardcoded editors and file managers. I'm using org.freedesktop.FileManager1 as a fallback if none of the file managers are found

File: flathub.json (hunk 2 above)
Comment (kyle-sylvestre) [reply flathub.json:2]: Only building x64 for the demo right now, might expand architectures if this flatpak gets enough traction. What yml changes are needed to install x64 and aarch64 binaries?

**Pull request closed without merging**